forge-core/src/main/java/forge/card/CardEdition.java

```diff
@@ -165,41 +165,15 @@ public static List<String> getNames() {
         }
     }
 
-    public static class CardInSet implements Comparable<CardInSet> {
-        public final CardRarity rarity;
+    public static class InSet {
```

Comment:
I wonder if we should be pulling these out of this file now that there's 3 subclasses? Also, I was looking at this briefly last month, and IntelliJ suggests converting these to `Record` types. Which removes a lot of this boilerplate.

Reply:
Ah interesting 🤔 

If it works for Android, I can try to change them into records (with or without base class?)

It's just that I need to find a new place for the CollectorNumber function

Reply:
changed into records

Reply:
Records might not work with Android 11 (my phone, for example)

@Agetian can you check the Android build for this?

Reply:
@tehdiplomat looks like the android build tools can backport it? Or is that something you can only test with runtime?

Reply:
Lets just pull it out of this PR for now. We can run a different PR with the upgrade later and fool around with how functional it is in android.